Multiserver

fatkudu opened the pull request:
[AsciiDoc](http://asciidoc.org/) port of _Multi-Server Install Guide_.  This PR covers pre-8.7.6 conversion of the old documentation.  The 8.7.6 updates will be submitted via a separate PR once this one has been approved and merged.

Still waiting on feedback as to whether-or-not the following documentation should be retained:

* Non-smartphone support
* Zimbra Connector for Blackberry Enterprise Server (BES)
* Touch Client? Still supported? Deprecate?

Also waiting on feedback as to the browser support info for web & admin clients. If we don't get this information (very) soon, I recommend we merge this PR as-is so that we can write and add the 8.7.6 updates.  We can continue to evolve this and the other documentation over time.

Commit:
PDF for multi-server install

Downloaded from the link labeled:
    **Multi-Server Installation Guide**
In the column headed:
    **Zimbra Collaboration 8.7 – Network Edition section**
from https://www.zimbra.com/documentation/

Commit:
Very rough .adoc source

Created by cutting the contents of a PDF using **Preview**
on Mac and pasting the contents in LibreOffice, then
saving/exporting the result to HTML

Commit:
Use {product-name}

Replace uses of Zimbra Collaboration with {product-name}

Commit:
Delete TOC and Index

Commit:
ZCS references replaced

Replaced ZCS references with {product-name} or {product-abbrev}.
Generally preferred {product-name} for consistency of terminology.

Commit:
Rename basicconfig.adoc as installconfig.adoc

Plan to use much of the same content for single and multi-server

Commit:
New basicconfig.adoc

This is now just a preamble plus an include of installconfig.adoc
which is shared between single and multi server install docs

Commit:
multiserver/single server merged installconfig.adoc

1. Split into multiple sections/tables, one for each menu
2. Added a first column for *Server Configured* from the multiserver
   guide
3. Re-organised sections slightly to be more aligned with order in
   install script.
4. Added some missing info
5. Removed documentation for `r) Start servers after configuration`
   which is no longer in the installer (well, commented out...)
6. Added info that was only in the multiserver guide.  Why single
   sourcing is a good idea...

Commit:
Added networkeditiondoc=TRUE

This enables some conditional includes so that we could at least
theoretically have different FOSS and NE docs.

Commit:
Multi-Server wording added/Conditional inclusion

Commit:
Most content moved to **packages.adoc**

Most of what was in here is content shared between single and
multi-server docs.  Moved that content to packages.adoc

Commit:
packages.adoc - content from overviewinstallprocess.adoc

This is shared between single and multi server docs

Commit:
multi-server-install-intro.adoc

Commit:
scanningattachments.adoc multi server changes

Commit:
images/zimbra-proxy.png

Picture explaining the positions of Zimbra Proxy and its relationships to other
components of Zimbra Collaboration.

Picture was corrupt in the original 8.7 multiserver PDF.  Got this
from the 8.6 PDF and cropped the result using Preview

Commit:
multi-server-install-planning.adoc

Commit:
Partial multi-server-install.adoc

Up to the end of the section on "Planning for the Installation"

Commit:
Use {product-version}

Commit:
multiserver - use settings.adoc/product-version

Commit:
Re-apply upstream changes to packages.adoc

Much of the content now in packages.adoc was in overviewinstallprocess.adoc
Re-applied upstream changes from there.
e.g. More use of {product-short} / {product-name} / {product-edition-commercial}
     Use single '*' for bold.

Commit:
multi-server doc originated changes

Commit:
multi-server-install.adoc includes prerequisites.adoc

This now has the same ordering as was in the single server install.
May split prerequisites.adoc up so that this stuff is at the end
of the doc as it is in the original multi-server pdf.

Commit:
prerequisites.adoc

gordyt committed:
Delete duplicate attributes

gordyt committed:
Added section: Configuring LDAP Replication

Commit:
prerequisites->prepareserver & systemrequirements

Renaming prerequisites.adoc prepareserver.adoc
Also copying to systemrequirements.adoc as plan to split the doc in 2
and move the requirements to the end of the install guides like the
original multi-server guide.

Commit:
Split prepareserver.adoc systemrequirements.adoc

These were copies of each other.  Now split into
the bits we intended for each.

Commit:
Re-arrange stuff under Preparing...

Re-arranging these sections to better mirror the multi-server
installguide.
Now include prepareserver.adoc, which includes:
* dnsconfig.adoc
* hifidocpreview.adoc
* modosconfig.adoc

systemrequirements.adoc now at the end

Commit:
modosconfig.adoc less content & now sub-section

* Modifying Operating System Configurations now a sub-section of
  Preparing Your Server Environment
* Commented out section talking about OS specific mods documented
  in docs that no longer exist.  Kept for reference in case
  we decide it is worth re-introducing them, with links to
  archive docs.

Commit:
Move sections down a level

Move sections under **Preparing Your Server Environment**
like they are in the multi-server document.

Commit:
dnsconfig.adoc added text from multi-server PDF

gordyt committed:
Extracted LDAP error code table to include

Pulled part of the section related to monitoring
LDAP replication status out of the multi-server-ldap-replication
section and put that into its own file so that it could be
included in the multi-master LDAP section of the documentation.

gordyt committed:
Added section on multi-master LDAP config

gordyt committed:
Adding Mailbox Server to Single-Server Config

Added new section.

Commit:
[[main_menu_options]] target added

Commit:
Partial 'Multiple-Server Installation' section

Commit:
Link in new section

gordyt committed:
multiserver - 'post install tasks' section

- Also the Unintall section

gordyt committed:
multiserver - 'final setup' section

gordyt committed:
multiserver - 'zimbra snmp' section

gordyt committed:
multiserver - 'zimbra archiving' section

Commit:
multiserver - 'Installing the MTA server' section

gordyt committed:
multiserver - 'installing zimbra proxy' section

gordyt committed:
Updated missing multiserver install cross-references

gordyt committed:
Tweaks for styling consistency, fixups

gordyt committed:
Add multi-server build to README

NOTE: PDF build is not currently working.

Commit:
Delete Working doc and original 8.7 PDF

gordyt commented:
+1

plobbes commented on index-single-server-install.adoc:
```diff
@@ -1,9 +1,11 @@
+:networkeditiondoc: TRUE
 = {product-name} {document-title}
-v8.7, April 2016
+v{product-version}, April 2016
```
Should the date should be a variable that lives along side of the version?

fatkudu replied:
Done.

plobbes commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,240 @@
+  that is installed on the single server node.
+* A copy of the {product-abbrev} license needs to be added to a
+  directory on the new machine.
+* you are adding {product-short} Proxy to {product-abbrev}, this
```
nit: you → You and remove 1 extra leading space from the two lines below.

fatkudu replied:
Done

plobbes commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,240 @@
+
+* {product-short} Mailbox Server is installed on the prepared machine.
+* Customized configuration for the single-server, such as custom themes
+and Zimlets are added to the new mailbox server.
```
nit: indent this line for consistency with other lines

fatkudu replied:
Done

plobbes commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,240 @@
+. From the Main menu, type `2` to go to the Store configuration
+menu.
+** Type `4` and set the password for the administrator account. This
+should be the same password as configured on the original
```
nit: inconsistent use of leading whitespace on this and the next few lines compared to other parts of the list

fatkudu replied:
Done.

plobbes commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,240 @@
+----
+zmprov -l gs <singleserver.com> | grep -i serviceenabled
+----
+
```
nit: remove blank lines here to the end?

fatkudu replied:
Done.

plobbes replied:
suggestion: change the example to use zmprov as it can be used (piping to grep isn't needed here)...
`zmprov -l gs <singleserver.com> zimbraserviceenabled`

plobbes commented on systemrequirements.adoc:
```diff
@@ -258,6 +277,11 @@ including:
 
 * Various device/operating system combinations with mobile WAP browser.
```
Seems like this whole non-smartphone section should go.  AFAIK, we don't support non-smartphones at this point. WAP in general is dead and I don't recall Zimbra supporting it anyway.  Also that link (non-https by the way) redirects to a generic page about ZCS... nothing mobile specific.

gordyt replied:
Removed non-smartphone section.

plobbes commented on systemrequirements.adoc:
```diff
@@ -258,6 +277,11 @@ including:
+
+ifdef::networkeditiondoc[]
 [[Zimbra_Touch_Client]]
 == {product-short} Touch Client ({product-edition-commercial} Only)
```
Wonder if PM can confirm that the touch client is even supported anymore.

fatkudu replied:
It is certainly present in 8.7 - which is what we're reviewing here.

plobbes commented on systemrequirements.adoc:
```diff
@@ -266,13 +290,29 @@ including:
+endif::networkeditiondoc[]
+
+ifdef::networkeditiondoc[]
+[[BES]]
```
I thought BES is dead (our last release was for 7.2.x, no?).  Seems like this should be removed.

plobbes reviewed (commented):
A number of comments...

Commit:
New variable product-release-date

Commit:
indentation/white space/You not you

grishick commented on basicconfig.adoc:
```diff
@@ -2,262 +2,9 @@
+The default configuration installs Zimbra-LDAP, Zimbra-MTA with
+anti-virus and anti-spam protection, the Zimbra mailbox server, SNMP
+monitoring tools (optional), Zimbra-spell (optional), the logger tool
+(optional), and the Zimbra proxy (optional) on one server.
```
why is {product-short} place holder replaced with "Zimbra" here?

gordyt replied:
This has been updated.

grishick commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,228 @@
+
+== Setup Requirements For Adding a Mailbox Server
+
+* The new machine you are adding must have the same operating system,
```
I wonder if the statement about having to have the same operating system is actually true and if there is any reason for this strict requirement (latest version, patch levels). I suspect it is possible to run Zimbra-LDAP on CentOS7, zimbra-store on Ubuntu16 and zimbra-proxy on Ubuntu14 in the same multi-server set up without any issues.

gordyt replied:
Greg should that part be yanked?  Are we comfortable with removing that restriction from our published docs?

grishick commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,228 @@
+== Configuring the Mailbox Server
+
+The host name and zmhostname configured on the mailbox server is the
+same as on the name on the single server.
```
"is the same as on the name on the single server" - there are some extraneous words here. Perhaps, it should be "is the same as on the single server" or "are the same as on the single server"

gordyt replied:
This has been fixed

grishick commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,228 @@
+. Type `r` to return to the Main menu.
+. From the Main menu, type `2` to go to the Store configuration
+menu.
+** Type `4` and set the password for the administrator account. This
```
I don't think this is correct. When you are installing a second mailstore node you should choose to NOT create admin user, because admin user has already been created when you installed the first mailstore. So, I think, that instead of "4 - set admin password", people should be choosing "2" to set "create admin user" to "no"

gordyt replied:
This has been fixed.

grishick commented on multi-server-adding-mailbox.adoc:
```diff
@@ -0,0 +1,228 @@
+The host name and zmhostname configured on the mailbox server is the
+same as on the name on the single server.
+
+Make sure you know the LDAP master password as you configure it on the
```
If you are installing zmproxy or MTA on the second node, you will also need to write down "Bind password for postfix ldap user", "Bind password for amavis ldap user" and "Bind password for nginx ldap user" - all these are already in LDAP if these services have been installed on the first node.

gordyt replied:
This has been fixed.

grishick commented on multi-server-install-main.adoc:
```diff
@@ -0,0 +1,1716 @@
+Found zimbra-convertd
+Found zimbra-memcached
+Found zimbra-proxy
+Found zimbra-archiving
```
8.7.6 also has zimbra-chat and zimbra-drive

grishick commented on multi-server-install-main.adoc:
```diff
@@ -0,0 +1,1716 @@
+Install zimbra-proxy [Y] n
+
+Install zimbra-archiving [N] n
+Checking required space for zimbra-core
```
zimbra-chat and zimbra-drive are now also available for 8.7.6

gordyt committed:
Removed non-smartphone support section

gordyt committed:
replace Zimbra references with {product-short}

gordyt committed:
multi-server-adding-mailbox updates

- updated awkward wording
- do not create admin user
- record additional passwords if necessary
